scripts/linux/consolidate-build-artifacts.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+###############################################################################
+# This script sets up the docker build context for various entities.
+
+# When copying rust artifacts into the build context we try to mimic the
+# directory structure of the target folder.
+###############################################################################
+
+###############################################################################
+# Define Environment Variables
+###############################################################################
+# Get directory of running script
+DIR=$(cd "$(dirname "$0")" && pwd)
+
+BUILD_REPOSITORY_LOCALPATH=${BUILD_REPOSITORY_LOCALPATH:-$DIR/../..}
+BUILD_BINARIESDIRECTORY=${BUILD_BINARIESDIRECTORY:-$BUILD_REPOSITORY_LOCALPATH/target}
+SCRIPT_NAME=$(basename "$0")
+
+###############################################################################
+# Print usage information pertaining to this script and exit
+###############################################################################
+function usage() {
+    echo "$SCRIPT_NAME [options]"
+    echo ""
+    echo "options"
+    echo " --artifact-name           The name of the artifact to consolidate for which to consolidate the artifacts."
+    echo " --build-binaries-dir      The name of the artifact to consolidate for which to consolidate the artifacts."
+    echo " -h, --help                Print this help and exit."
+    exit 1
+}
+
+function print_help_and_exit() {
+    echo "Run $SCRIPT_NAME --help for more information."
+    exit 1
+}
+
+###############################################################################
+# Obtain and validate the options supported by this script
+###############################################################################
+function process_args() {
+    save_next_arg=0
+    for arg in "$@"; do
+        if [ ${save_next_arg} -eq 1 ]; then
+            ARTIFACT_NAME=$arg
+            save_next_arg=0
+        elif [ ${save_next_arg} -eq 2 ]; then
+            BUILD_BINARIESDIRECTORY=$arg
+            save_next_arg=0
+        else
+            case "$arg" in
+            "-h" | "--help") usage ;;
+            "--artifact-name") save_next_arg=1 ;;
+            "--build-binaries-dir") save_next_arg=2 ;;
+            *) usage ;;
+            esac
+        fi
+    done
+}
+
+process_args "$@"
+
+cd $BUILD_REPOSITORY_LOCALPATH
+
+TARGET_ARM32V7="armv7-unknown-linux-gnueabihf"
+TARGET_ARM64V8="aarch64-unknown-linux-gnu"
+TARGET_AMD64_MUSL="x86_64-unknown-linux-musl"
+
+case "$ARTIFACT_NAME" in
+generic-mqtt-tester)
```

Comment:
In the script you allow the caller to specify which artifact they want to consolidate, but then the only caller (the yaml file) always calls the script twice, once for each artifact. So do we really need the extra logic to just consolidate one artifact or the other?

Reply:
If we are using it for local builds you won't want to call it for both, because the artifacts for both might not be available.